### PR TITLE
Simplify creating and unrolling symmetric tensors

### PR DESCRIPTION
**Pull request by gassmoeller**
Something I noticed while reviewing #4370. We have quite a number of places in the code where we create Tensors from arrays or vectors, or unroll Tensors into arrays or vectors. We could simplify this using Tensor constructors, and appropriate `unroll` functions. Such functions are already provided by deal.II for the `Tensor` class, but they dont exist for `SymmetricTensor`. I here created small utility functions for `SymmetricTensor`, but of course it would be nice to use a version inside deal.II. Would that be of interest?

I can extend this PR to simplify a lot of places in ASPECT with the utility functions, but I first wanted to ask for feedback if the current interface is correct and if this is useful overall.

### CODE REVIEW

**Comment by tjhei** [comment]
I think it is great to simplify and clean up the code based like this.
I already mentioned in the other PR that I don't like the two-step creation of an ArrayView. We could use an iterator version or not? You can also make a simplified version that only takes a begin iterator. Do you want me to try that out?

File: include/aspect/utilities.h [hunk 1]
```diff
@@ -1245,6 +1245,33 @@ namespace aspect
...
+       * Unroll a SymmetricTensor into a series of doubles.
+       */
+      template <int dim>
+      inline
+      void
+      unroll(const SymmetricTensor<2,dim> &tensor,
+             const ArrayView<double> &values)
+      {
+        for (unsigned int i=0; i < SymmetricTensor<2,dim>::n_independent_components; ++i)
+          values[i] = tensor[SymmetricTensor<2,dim>::unrolled_to_component_indices(i)];
```
Comment (bangerth) [inline include/aspect/utilities.h:1273]: Can you add
```
  Assert (values.size() == ...n_indep_comp, ...)
```
?

Same for the other function.

**Review by bangerth — COMMENTED** [review]
Yes, this is good. In an ideal world, the two new functions would live inside `dealii::SymmetricTensor`. I would gladly accept a patch.

**Comment by gassmoeller** [comment]
Ok, I think I addressed all comments. The new versions have more expressive names and use iterators instead of an ArrayView, which makes them more flexible to use with different data containers. Feel free to take a look and when this is merged I can prepare a similar path for deal.II.

**Review by bangerth — APPROVED** [review]

**Pull request merged**